recommender/recommender.py

```diff
@@ -657,43 +678,106 @@ def endorse_resource(self, data, _suffix=''):
-                result['error']: the error message generated when the deendorsement fails
-                result['recommendation']: (Dict) the deendorsed resource
-                result['recommendation']['reason']: the reason why the resouce was deendorsed
+                result['Success']: the boolean indicator for whether the removal is complete
+                result['error']: the error message generated when the removal fails
+                result['recommendation']: (Dict) the removed resource
+                result['recommendation']['reason']: the reason why the resouce was removed
         """
-        # TODO: this function was named delete_resource previously, which should be changed in test_recommender
         if not self.get_user_is_staff():
-            msg = 'Deendorse resource without permission'
+            msg = 'Remove resource without permission'
             return self.error_handler(msg, 'deendorse_resource')
-        resource_id = data['id']
-        idx = self.get_entry_index(resource_id, self.recommendations)
-        if idx not in range(0, len(self.recommendations)):
+
+        resource_id = stem_url(data['id'])
+        if resource_id not in self.recommendations:
             msg = 'The selected resource is not existing'
             return self.error_handler(msg, 'deendorse_resource', resource_id)
 
         result = {}
         result['id'] = resource_id
-        deendorsed_resource = deepcopy(self.recommendations[idx])
-        del self.recommendations[idx]
+        deendorsed_resource = deepcopy(self.recommendations[resource_id])
+        del self.recommendations[resource_id]
 
         deendorsed_resource['reason'] = data['reason']
-        self.deendorsed_recommendations.append(deendorsed_resource)
+        self.deendorsed_recommendations[resource_id] = deendorsed_resource
         result['Success'] = True
         result['recommendation'] = deendorsed_resource
         tracker.emit('deendorse_resource', result)
         return result
 
     @XBlock.json_handler
+    def export_resources(self, _data, _suffix):
+        """
+        Export all resources from the Recommender. This is intentionally not limited to staff
+        members (community contributions do not belong to the course staff). Sensitive
+        information is exported *is* limited (flagged resources, and in the future, PII if
+        any).
+        """
+        result = {}
+        result['export'] = {
+            'recommendations': self.recommendations,
+            'deendorsed_recommendations': self.deendorsed_recommendations,
+            'endorsed_recommendation_ids': self.endorsed_recommendation_ids,
+            'endorsed_recommendation_reasons': self.endorsed_recommendation_reasons,
+        }
+        if self.get_user_is_staff():
+            result['export']['flagged_accum_resources'] = self.flagged_accum_resources
+
+        result['Success'] = True
```

Comment:
You don't need "Success = True", just use the HTTP status code to indicate success.